app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,32 @@
+class ReviewsController < ApplicationController
+  before_action :require_login
+  before_action :merge_current_user, only: [:create]
+
+  def new
+    @review = Book.find(params[:book_id]).reviews.build
+  end
+
+  def create
+    @review = Book.find(book_id).reviews.build(review_params)
+    if @review.save
+      redirect_to book_path(book_id), notice: 'Thanks for adding your review'
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def review_params
+    params.require(:review).permit(:book_id, :body, :user_id)
+  end
+
+  def book_id
+    params[:review][:book_id]
+  end
+
+  def merge_current_user
+    # wtf can't put current_user.id as a hidden field in the form?
+    params[:review].merge! user_id: current_user.id
```

Comment:
I tried to use

```html
  <%= f.input :user_id, as: :hidden, value: current_user.id %>
```

in my form, but it would not render the value to the page, so I just set it here.